OpenRA.Mods.Common/Traits/Player/ProductionQueue.cs

```diff
@@ -433,7 +436,14 @@ public void Tick(PlayerResources pr)
 			{
 				var time = Queue.GetBuildTime(Item);
 				if (time > 0)
+				{
 					RemainingTime = TotalTime = time;
+					framesToSkip = TotalTime - TotalCost;
```

Comment:
How did you manage to comment on an outdated diff three hours ago when I last pushed more than two days ago?

Anyway, yeah, I can add a comment.

And yes, this will break again (ie. the original bug will reappear) if the difference between `TotalCost` and `TotalTime` gets large (if `TotalTime` ends up being more than twice the value of `TotalCost`). With the way our queues are set up atm, that will happen with a `CustomBuildTimeValue` that is five times that of `Valued.Cost`.

The problem is that integer division gets very imprecise when the two values are close to each other, so the result ends up being 1, which means that every frame would be skipped until they are all accounted for. Only then the regular cash ticks would occur, just like in the original bug.

Not sure how to do it any other way, though. We can't use proper floating-point division, since it's simulation code, and it also wouldn't make sense, as half a tick doesn't quite work ;)